write.js

```
@@ -11,15 +11,17 @@ class ChunkStoreWriteStream extends stream.Writable {
     chunkLength = Number(chunkLength)
     if (!chunkLength) throw new Error('Second argument must be a chunk length')
 
-    this._blockstream = new BlockStream(chunkLength, { zeroPadding: false })
+    const zeroPadding = opts.zeroPadding === undefined ? false : opts.zeroPadding
+    this._blockstream = new BlockStream(chunkLength, { zeroPadding })
     this._outstandingPuts = 0
 
     let index = 0
     const onData = chunk => {
       if (this.destroyed) return
 
       this._outstandingPuts += 1
-      store.put(index, chunk, () => {
+      store.put(index, chunk, (err) => {
+        if (err) return this.destroy(err)
```

Not sure if `.destroy` is the right thing

---

But this error was getting swallowed before which made debugging tricky